### PR TITLE
Add directions for using Azure Container Registry.

### PR DESCRIPTION
**Pull request by brendandburns**
@colemickens @ahmetalpbalkan

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/1734)

### CODE REVIEW

File: docs/user-guide/images.md [hunk 1]
```diff
@@ -100,6 +101,23 @@ Troubleshooting:
...
+### Using Azure Container Registry (ACR)
+When using Azure container registry you can authenticate using either an admin user or a service principal.
+In either case, authentication is done via standard Docker authentication.  These instructions assume the
+python based [az](https://github.com/azure/azure-cli) tooling.
+`
+```console
+# To generate new credentials
+az acr credential renew -n ${CLUSTER_NAME} -g ${CLUSTER_RESOURCE_GROUP}
+
+# To show existing credentials
```
Comment (colemickens) [inline docs/user-guide/images.md]: It doesn't seem to be rendering this correctly on the netlify preview... https://deploy-preview-1734--kubernetes-io-vnext-staging.netlify.com/docs/user-guide/images/#using-azure-container-registry-acr

I think you need to remove the extra backtick on line 108.

File: docs/user-guide/images.md [hunk 2]
```diff
@@ -100,6 +101,23 @@ Troubleshooting:
   - `plugins.go:56] Registering credential provider: aws-ecr-key`
   - `provider.go:91] Refreshing cache for provider: *aws_credentials.ecrProvider`
 
+### Using Azure Container Registry (ACR)
+When using Azure container registry you can authenticate using either an admin user or a service principal.
```
Comment (ahmetalpbalkan) [inline docs/user-guide/images.md]: nit: `Azure Container Registry, ...`

File: docs/user-guide/images.md [hunk 3]
```diff
@@ -100,6 +101,23 @@ Troubleshooting:
...
+In either case, authentication is done via standard Docker authentication.  These instructions assume the
+python based [az](https://github.com/azure/azure-cli) tooling.
+`
+```console
+# To generate new credentials
+az acr credential renew -n ${CLUSTER_NAME} -g ${CLUSTER_RESOURCE_GROUP}
```
Comment (ahmetalpbalkan) [inline docs/user-guide/images.md]: This command will not work if ACR registry is configured to use a Service Principal (recommended). It assumes ACR Admin Account is used (which might not be available when it goes GA). IMO this documentation should just refer to ACR docs which documents: 1) how to create a registry 2) create a SP account 3) make RBAC assignment for SP to use ACR.

File: docs/user-guide/images.md (hunk 2 above)
Comment (ahmetalpbalkan) [inline docs/user-guide/images.md]: oh, also please link to ACR website like others did.

**Comment by thockin** [comment]
I am assigned, but in name only.  Brendan will do tech LGTM and Devin (or a delegatee thereof) will do docs.

File: docs/user-guide/images.md [hunk 4]
```diff
@@ -100,6 +101,23 @@ Troubleshooting:
   - `plugins.go:56] Registering credential provider: aws-ecr-key`
   - `provider.go:91] Refreshing cache for provider: *aws_credentials.ecrProvider`
 
+### Using Azure Container Registry (ACR)
+When using Azure container registry you can authenticate using either an admin user or a service principal.
+In either case, authentication is done via standard Docker authentication.  These instructions assume the
+python based [az](https://github.com/azure/azure-cli) tooling.
```
Comment (devin-donnelly) [inline docs/user-guide/images.md]: "The following instructions assume you're using the `azure-cli` command-line tool."

File: docs/user-guide/images.md (hunk 1 above)
Comment (devin-donnelly) [inline docs/user-guide/images.md]: Correct. This is rendering as an H1 in mid-document; please check your formatting.

File: docs/user-guide/images.md (hunk 2 above)
Comment (brendandburns) [inline docs/user-guide/images.md]: done.

File: docs/user-guide/images.md (hunk 1 above)
Comment (brendandburns) [inline docs/user-guide/images.md]: fixed (I think)

File: docs/user-guide/images.md (hunk 4 above)
Comment (brendandburns) [inline docs/user-guide/images.md]: done.

File: docs/user-guide/images.md [hunk 5]
```diff
@@ -100,6 +101,23 @@ Troubleshooting:
...
+`
```
Comment (brendandburns) [inline docs/user-guide/images.md]: removed.

File: docs/user-guide/images.md (hunk 3 above)
Comment (brendandburns) [inline docs/user-guide/images.md]: directions removed, link added (though that doc is super verbose and wordy)

File: docs/user-guide/images.md [hunk 6]
```diff
@@ -100,6 +101,23 @@ Troubleshooting:
...
+```console
+# To generate new credentials
+az acr credential renew -n ${CLUSTER_NAME} -g ${CLUSTER_RESOURCE_GROUP}
+
+# To show existing credentials
+az acr credential show -n ${CLUSTER_NAME} -g ${CLUSTER_RESOURCE_GROUP}
+```
+
+Once you have those credentials you can [configure a Kubernetes Secret and use it to deploy a Pod]
+(http://kubernetes.io/docs/user-guide/images/#specifying-imagepullsecrets-on-a-pod)
```
Comment (brendandburns) [inline docs/user-guide/images.md]: done.

**Comment by brendandburns** [comment]
comments addressed, please re-check.

Thanks!
--brendan

**Comment by brendandburns** [comment]
@devin-donnelly can I get a review on this?

Thanks!
--brendan

File: docs/user-guide/images.md [hunk 7]
```diff
@@ -100,6 +101,25 @@ Troubleshooting:
...
+You first need to create a registry and generate credentials, complete documentation for this can be found in
+the [Azure container registry documentation](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-get-started-azure-cli)
+
+Once you have created your container registry, you will use the following credentials to login:
+   * `DOCKER_USER` : service principal, or admin username
+   * `DOCKER_PASSWORD`: service principal password, or admin user password
+   * `DOCKER_REGISTRY_SERVER`: `${some-registry-name}.azurecr.io`
+   * `DOCKER_EMAIL`: `${some-email-address}`
+
+Once you have those variables filled in you can [configure a Kubernetes Secret and use it to deploy a Pod]
```
Comment (devin-donnelly) [inline docs/user-guide/images.md]: This link does not render correctly. Check to see if you don't have a whitespace issue.

File: docs/user-guide/images.md [hunk 8]
```diff
@@ -100,6 +101,25 @@ Troubleshooting:
   - `plugins.go:56] Registering credential provider: aws-ecr-key`
   - `provider.go:91] Refreshing cache for provider: *aws_credentials.ecrProvider`
 
+### Using Azure Container Registry (ACR)
+When using [Azure Container Registry](https://azure.microsoft.com/en-us/services/container-registry/)
+you can authenticate using either an admin user or a service principal.
+In either case, authentication is done via standard Docker authentication.  These instructions assume the
+[azure-cli](https://github.com/azure/azure-cli) command line tool.
+
+You first need to create a registry and generate credentials, complete documentation for this can be found in
```
Comment (devin-donnelly) [inline docs/user-guide/images.md]: Replace comma with period; capitalize "Complete."

File: docs/user-guide/images.md [hunk 9]
```diff
@@ -100,6 +101,25 @@ Troubleshooting:
...
+the [Azure container registry documentation](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-get-started-azure-cli)
```
Comment (ahmetalpbalkan) [inline docs/user-guide/images.md]: missing period at the end.

File: docs/user-guide/images.md (hunk 8 above)
Comment (brendandburns) [inline docs/user-guide/images.md]: done.

File: docs/user-guide/images.md (hunk 7 above)
Comment (brendandburns) [inline docs/user-guide/images.md]: done.

**Comment by brendandburns** [comment]
@devin-donnelly comments addressed, please re-check.

Thanks!
--brendan

**Comment by ahmetalpbalkan** [comment]
I think this is good to go.

**Comment by brendandburns** [comment]
Ping @devin-donnelly

**Pull request merged**